src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/prisonperson/profiledetails/ProfilesDetailsIntTest.kt

```diff
@@ -633,14 +643,29 @@ class ProfilesDetailsIntTest : IntegrationTestBase() {
           }
         }
 
-        assertThrows<BadDataException> {
-          service.upsertProfileDetails("A1234AA", UpsertProfileDetailsRequest("BUILD", "UNKNOWN_CODE"))
-        }.also {
-          assertThat(it.message).contains("UNKNOWN_CODE")
-        }
+        webTestClient.upsertProfileDetails("A1234AA", "BUILD", "UNKNOWN_CODE")
+          .expectStatus().isBadRequest
+          .expectErrorContaining("UNKNOWN_CODE")
       }
     }
 
     private fun findBooking(bookingId: Long = booking.bookingId) = offenderBookingRepository.findByIdOrNull(bookingId)!!
+
+    fun WebTestClient.upsertProfileDetailsOk(offenderNo: String, profileType: String, profileCode: String?) =
+      upsertProfileDetails(offenderNo, profileType, profileCode)
+        .expectStatus().isOk
+        .expectBody<UpsertProfileDetailsResponse>()
+
+    fun WebTestClient.upsertProfileDetails(offenderNo: String, profileType: String, profileCode: String?) =
+      put().uri("/prisoners/$offenderNo/profile-details")
+        .headers(setAuthorisation(roles = listOf("ROLE_NOMIS_PRISON_PERSON")))
+        .bodyValue(UpsertProfileDetailsRequest(profileType, profileCode))
+        .exchange()
+
+    fun WebTestClient.ResponseSpec.expectErrorContaining(partialMessage: String) =
+      expectBody<ErrorResponse>()
+        .consumeWith {
+          assertThat(it.responseBody!!.userMessage).contains(partialMessage)
+        }
```

Comment:
Hmm not sure which i prefer, this or something like:
.expectBody().jsonPath("$.userMessage").value<String> {
   assertThat(it).contains(partialMessage)
}